src/bytes_mut.rs

```diff
@@ -1064,6 +1064,17 @@ impl Buf for BytesMut {
             cnt,
             self.remaining(),
         );
+
+        // Advancing by the length is the same as resetting the length to 0,
+        // except this way we get to reuse the full capacity.
+        if cnt == self.len {
+            unsafe {
+                // SAFETY: `advance` is allowed to reuse the allocation.
+                self.set_len(0);
+            }
```

Comment:
The safety requirement for `set_len` is that the length must not be greater than the capacity, so that is what the safety comment should address. The comment you have right now could stay, but not as a safety comment.
```suggestion
            // SAFETY: Zero is not greater than the capacity.
            unsafe { self.set_len(0) };
```
Also, safety comments should generally go before the unsafe block. I know there are some places where we put it inside, but they should be fixed at some point.

Reply:
> The safety requirement for set_len is that the length must not be greater than the capacity, so that is what the safety comment should address.

:+1: I'll include it soon.

Reply:
Notice that the semicolon is outside the unsafe block. That way it fmts on one line.